### PR TITLE
add dedot plugin

### PR DESCRIPTION
**Pull request by golgoth31**
Add a dedot plugin to remove dot into field name, almost useful for elasticsearch

### CODE REVIEW

**Comment by golgoth31** [comment]
Hello,
I've updated this PR to mv dedot plugin into elasticsearch only images, bump the elasticsearch client to 7.4.0 and bump the elasticsearch fluent plugin to 4.0.
Is there any problem to merge this ?

File: templates/Gemfile.erb [hunk 1]
```diff
@@ -21,15 +21,16 @@ gem "fluent-plugin-secure-forward"
 gem "fluent-plugin-record-reformer"
 gem "fluent-plugin-rewrite-tag-filter"
 <% else %>
-gem "fluent-plugin-rewrite-tag-filter", "~> 2.2.0"
+gem 'fluent-plugin-rewrite-tag-filter', '~> 2.2'
```
Comment (repeatedly) [inline templates/Gemfile.erb]: Don't touch other plugin.

Comment (golgoth31) [reply templates/Gemfile.erb]: ok, I revert this :)

File: templates/Gemfile.erb [hunk 2]
```diff
@@ -27,9 +27,10 @@ gem "fluent-plugin-rewrite-tag-filter", "~> 2.2.0"
 gem "elasticsearch", "~> 6.0"
 gem "fluent-plugin-elasticsearch", "~> 3.5.5"
 <% when "elasticsearch7" %>
-gem "elasticsearch", "~> 7.0"
-gem "fluent-plugin-elasticsearch", "~> 3.7.1"
+gem "fluent-plugin-dedot_filter", "~> 1.0"
+gem "elasticsearch", "~> 7.4.0"
```
Comment (ganmacs) [inline templates/Gemfile.erb]: Why did you change this line? dedot_filter seems not to require this version. https://github.com/lunardial/fluent-plugin-dedot_filter

Besides, No need to change since `~> 7.0` installs v7.4.0. see https://guides.rubygems.org/patterns/

File: templates/Gemfile.erb [hunk 3]
```diff
@@ -27,9 +27,10 @@ gem "fluent-plugin-rewrite-tag-filter", "~> 2.2.0"
...
 gem "elasticsearch-xpack", "~> 7.4.0"
+gem "fluent-plugin-elasticsearch", "~> 4.0"
```
Comment (ganmacs) [inline templates/Gemfile.erb]: ditto

File: templates/Gemfile.erb (hunk 2 above)
Comment (golgoth31) [reply templates/Gemfile.erb]: dedot plugin was not installed before this patch, the version 1.0 is working with fluent >=0.14

I will correct the deployed version of elasticsearch gem

File: templates/Gemfile.erb (hunk 3 above)
Comment (golgoth31) [reply templates/Gemfile.erb]: this is the latest release available https://rubygems.org/gems/fluent-plugin-elasticsearch

Comment (ganmacs) [reply templates/Gemfile.erb]: Yes. but this PR is for installing dedot plugin. Updating elasticsearch is not related unless dedot plugin needs the version of elasticsearch.

File: docker-image/v1.8/debian-elasticsearch7/Gemfile.lock [hunk 4]
```diff
@@ -124,11 +126,12 @@ PLATFORMS
...
 DEPENDENCIES
   elasticsearch (~> 7.0)
-  elasticsearch-xpack (~> 7.4.0)
+  elasticsearch-xpack (~> 7.0)
   ffi
   fluent-plugin-concat (~> 2.4.0)
+  fluent-plugin-dedot_filter (~> 1.0)
   fluent-plugin-detect-exceptions (~> 0.0.12)
-  fluent-plugin-elasticsearch (~> 3.7.1)
+  fluent-plugin-elasticsearch (~> 4.0)
```
Comment (cosmo0920) [inline docker-image/v1.8/debian-elasticsearch7/Gemfile.lock]: Contaminated updating Elasticsearch plugin.
We shouldn't touch plugins and dependent gems versions.

File: docker-image/v1.8/debian-elasticsearch7/Gemfile [hunk 5]
```diff
@@ -14,8 +14,9 @@ gem "fluent-plugin-record-modifier", "~> 2.0.0"
 gem "fluent-plugin-detect-exceptions", "~> 0.0.12"
 gem "fluent-plugin-rewrite-tag-filter", "~> 2.2.0"
 gem "elasticsearch", "~> 7.0"
-gem "fluent-plugin-elasticsearch", "~> 3.7.1"
-gem "elasticsearch-xpack", "~> 7.4.0"
+gem "fluent-plugin-elasticsearch", "~> 4.0"
+gem "elasticsearch-xpack", "~> 7.0"
```
Comment (cosmo0920) [inline docker-image/v1.8/debian-elasticsearch7/Gemfile]: elasticsearch-xpack shouldn't be downgraded.
Your patch sometimes contains contaminated upgrading/downgrading.

File: docker-image/v1.8/debian-elasticsearch7/Gemfile.lock [hunk 6]
```diff
@@ -124,11 +126,12 @@ PLATFORMS
 
 DEPENDENCIES
   elasticsearch (~> 7.0)
-  elasticsearch-xpack (~> 7.4.0)
+  elasticsearch-xpack (~> 7.0)
```
Comment (cosmo0920) [inline docker-image/v1.8/debian-elasticsearch7/Gemfile.lock]: ditto.

File: docker-image/v1.8/debian-elasticsearch7/Gemfile.lock (hunk 4 above)
Comment (golgoth31) [reply docker-image/v1.8/debian-elasticsearch7/Gemfile.lock]: No problem, I revert it and I will push a new PR :)

File: docker-image/v1.8/debian-elasticsearch7/Gemfile (hunk 5 above)
Comment (golgoth31) [reply docker-image/v1.8/debian-elasticsearch7/Gemfile]: No, I just harmonized the elasticsearch-xpack version number with the one of elasticsearch plugin. I will make an other PR for this.

Comment (cosmo0920) [reply docker-image/v1.8/debian-elasticsearch7/Gemfile]: Make sense.

**Comment by repeatedly** [comment]
Thanks!
